### PR TITLE
Consist threading (on top of Flo's Phyics threading)

### PR DESCRIPTION
**Pull request by cam72cam**
This is an extension on top of the work done in https://github.com/TeamOpenIndustry/ImmersiveRailroading/pull/1372

It allows each distinct "consist" to be calculated in parallel, further breaking up the tasks required by IR's physics in each world.

Original Code:
PreWorldTick -> each world -> setup physics -> calculate consist -> send update

Flo's PR:
PreWorldTick -> each world -> submit physics thread job (no further impact on TPS)
each physics thread job -> setup physics -> calculate consist -> send update

This PR:
PreWorldTick -> each world -> setup physics -> submit consist thread jobs
(calling thread waits)
each consist thread job -> calculate consist
(calling thread continues)
combine consists -> send update

tl;dr:
Flo's PR shunts all of the physics, consist, and (depending on MC) update message serialization onto it's own per-world thread so it does not impact TPS
This PR calculates each consist in parallel (up to the configured thread limit) which means each physics iteration is faster.
Together it means no TPS impact + more capacity for lots of trains running at the same time.

### CODE REVIEW

**Comment by cam72cam** [comment]
Closing for now as too complex
